File: twc-flutter/lib/core/models/product.dart
class Product {
  final String id;
  final String sellerId;
  final String name;
  final String? description;
  final double price;
  final String? category;
  final List<String>? images;
  final String? mainImage;
  final double averageRating;
  final int reviewsCount;
  final bool? isFavorited;
  final DateTime createdAt;
  final String formattedPrice;

  Product({
    required this.id,
    required this.sellerId,
    required this.name,
    this.description,
    required this.price,
    this.category,
    this.images,
    this.mainImage,
    required this.averageRating,
    required this.reviewsCount,
    this.isFavorited,
    required this.createdAt,
    required this.formattedPrice,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
    // Laravel renvoie les id en nombre (int), pas en String : cast sûr avec .toString()
    id: json['id'].toString(),
    sellerId: json['seller_id'].toString(),
    name: json['name'],
    description: json['description'],
    price: double.tryParse('${json['price'] ?? 0}') ?? 0,
    category: json['category'],
    images: json['images'] != null ? List<String>.from(json['images']) : null,
    mainImage: json['main_image'],
    averageRating: double.tryParse('${json['average_rating'] ?? 0}') ?? 0,
    reviewsCount: json['reviews_count'] ?? 0,
    isFavorited: json['is_favorited'],
    createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
    formattedPrice: json['formatted_price'] ?? '',
  );
}

File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'seller_id', 'name', 'description', 'price', 'category', 'images',
        'video_url', 'location', 'contact_info', 'stock_quantity', 'is_active',
        'is_approved', 'views_count', 'featured', 'expires_at'
    ];

    protected $casts = [
        'images' => 'array',
        'price' => 'decimal:2',
        'is_active' => 'boolean',
        'is_approved' => 'boolean',
        'featured' => 'boolean',
        'stock_quantity' => 'integer',
        'expires_at' => 'datetime',
    ];

    protected $appends = ['main_image', 'formatted_price', 'is_out_of_stock', 'average_rating', 'reviews_count', 'favorites_count'];

    public function seller() { return $this->belongsTo(User::class, 'seller_id'); }
    public function orders() { return $this->hasMany(Order::class); }
    public function favorites() { return $this->hasMany(Favorite::class); }
    public function reviews() { return $this->hasMany(Review::class); }

    public function getImagesAttribute($value)
    {
        $images = is_array($value) ? $value : (json_decode($value ?? '[]', true) ?? []);
        return array_values(array_map(
            fn ($url) => \App\Support\MediaHelper::absoluteUrl($url),
            $images
        ));
    }

    public function getMainImageAttribute() { if ($this->images && is_array($this->images) && count($this->images) > 0) return $this->images[0]; return null; }
    public function getFormattedPriceAttribute() { return number_format($this->price, 0, ',', ' ') . ' CDF'; }
    public function getIsOutOfStockAttribute() { return $this->stock_quantity !== null && $this->stock_quantity <= 0; }
    public function getAverageRatingAttribute() { return $this->reviews()->avg('rating') ?? 0; }
    public function getReviewsCountAttribute() { return $this->reviews()->count(); }
    public function getFavoritesCountAttribute() { return $this->favorites()->count(); }

    public function scopeApproved($query) { return $query->where('is_approved', true); }
    public function scopeActive($query) { return $query->where('is_active', true); }
    public function scopeFeatured($query) { return $query->where('featured', true); }
    public function scopeSearch($query, $term) { return $query->where(function ($q) use ($term) { $q->where('name', 'LIKE', "%{$term}%")->orWhere('description', 'LIKE', "%{$term}%")->orWhere('category', 'LIKE', "%{$term}%"); }); }
    public function scopePriceRange($query, $min, $max) { return $query->whereBetween('price', [$min, $max]); }

    public function incrementViews() { $this->increment('views_count'); }
    public function isOwner($userId) { return $this->seller_id === $userId; }
}
